superduperdb/backends/base/query.py

```diff
@@ -145,6 +145,33 @@ def _set_db(r, db):
             parts.append((part, part_args, part_kwargs))
         self.parts = parts
 
+    def listeners(self, ):
+        listeners = self.db.show('listener')
+        vector_indices = self.db.show('vector_index')
+        dependencies = []
+        def _check_query_match(listener, query):
+            if (
+                listener.select.table_or_collection.identifier
+                == query.table_or_collection.identifier
+            ):
+                return True
+            return False
+
+        
+        for listener in listeners:
+
+            listener = self.db.listeners[listener]
+            if _check_query_match(listener, self):
+                dependencies.append({'type_id': 'listener', 'identifier': listener.identifier})
+
+        for vi in vector_indices:
```

Comment:
Don't follow this logic.

Reply:
Please provide doc-strings and comments. Name is odd, since called `listeners` but treats `Listener` and `VectorIndex`.

Reply:
doc strings will be last thing once we finalise logic